t/pubsub/pubsub.t

```diff
@@ -0,0 +1,215 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    if ((!defined $block->error_log) && (!defined $block->no_error_log)) {
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+});
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: setup route by serverless
+--- config
+    location /t {
+        content_by_lua_block {
+            local data = {
+                {
+                    url = "/apisix/admin/routes/pubsub",
+                    data = {
+                        plugins = {
+                            ["serverless-pre-function"] = {
+                                phase = "access",
+                                functions =  {
+                                    [[return function(conf, ctx)
+                                        local core = require("apisix.core");
+                                        local pubsub, err = core.pubsub.new()
+                                        if not pubsub then
+                                            core.log.error("failed to initialize pubsub module, err: ", err)
+                                            core.response.exit(400)
+                                            return
+                                        end
+                                        pubsub:on("cmd_kafka_list_offset", function (params)
+                                            return nil, "test"
+                                        end)
+                                        pubsub:wait()
+                                        ngx.exit(0)
+                                    end]],
+                                }
+                            }
+                        },
+                        uri = "/pubsub"
+                    },
+                },
+            }
+
+            local t = require("lib.test_admin").test
+
+            for _, data in ipairs(data) do
+                local code, body = t(data.url, ngx.HTTP_PUT, data.data)
+                ngx.say(code..body)
```

Comment:
Better not directly print the 201 status code, because when we rerun the test, code 200 will return instead.

Reply:
fixed